File: Farmsy/Core/FarmAxis.swift
import SwiftUI

/// The two taxonomy axes Aviah split out of the category list — "Type of place"
/// (`location_types`) and "How it is grown" (`methods`). Values are the
/// language-neutral ids stored server-side (validated against a closed list);
/// labels live here, client-side, exactly like FarmCategory. They combine with
/// the category filters, they do not replace them.
///
/// `organic` is deliberately absent from `methods`: it sits on both axes right
/// now with 20× different coverage (1,420 farms as a category vs 71 as a method),
/// so it stays in the category rail until the backend resolves that.
struct FarmAxisValue: Identifiable, Hashable {
    let id: String        // the stored id, e.g. "milk-tap"
    let label: String     // localized display label
    let icon: String      // SF Symbol, to match the quick-filter rows
}

enum FarmAxis {
    /// `farms.location_types`.
    static let placeTypes: [FarmAxisValue] = [
        .init(id: "shop",                   label: String(localized: "Farm shop"),        icon: "storefront"),
        .init(id: "vending-machine",        label: String(localized: "Vending machine"),  icon: "cabinet"),
        .init(id: "stall",                  label: String(localized: "Roadside stall"),   icon: "basket"),
        .init(id: "milk-tap",               label: String(localized: "Milk tap"),         icon: "drop"),
        .init(id: "self-picking",           label: String(localized: "Self-picking"),     icon: "hand.raised"),
        .init(id: "self-picking-unstaffed", label: String(localized: "Self-picking (unstaffed)"), icon: "hand.raised.slash"),
    ]

    /// `farms.methods` — organic intentionally excluded (see the type doc).
    static let methods: [FarmAxisValue] = [
        .init(id: "biodynamic",   label: String(localized: "Biodynamic"),   icon: "moon.stars"),
        .init(id: "regenerative", label: String(localized: "Regenerative"), icon: "arrow.3.trianglepath"),
        .init(id: "grass-fed",    label: String(localized: "Grass-fed"),    icon: "leaf"),
        .init(id: "sustainable",  label: String(localized: "Sustainable"),  icon: "globe.europe.africa"),
    ]
}

File: android/app/src/main/java/app/farmsy/android/core/FarmAxis.kt
package app.farmsy.android.core

import androidx.annotation.StringRes
import app.farmsy.android.R

/// The two taxonomy axes Aviah split out of the category list — "Type of place"
/// (location_types) and "How it's grown" (methods). Ids are language-neutral and
/// validated server-side; labels live client-side. They combine with the category
/// filters, they don't replace them. organic stays a category, not a method.
data class FarmAxisValue(val id: String, @StringRes val labelRes: Int)

object FarmAxis {
    val placeTypes = listOf(
        FarmAxisValue("shop", R.string.axis_shop),
        FarmAxisValue("vending-machine", R.string.axis_vending_machine),
        FarmAxisValue("stall", R.string.axis_stall),
        FarmAxisValue("milk-tap", R.string.axis_milk_tap),
        FarmAxisValue("self-picking", R.string.axis_self_picking),
        FarmAxisValue("self-picking-unstaffed", R.string.axis_self_picking_unstaffed),
    )

    val methods = listOf(
        FarmAxisValue("biodynamic", R.string.axis_biodynamic),
        FarmAxisValue("regenerative", R.string.axis_regenerative),
        FarmAxisValue("grass-fed", R.string.axis_grass_fed),
        FarmAxisValue("sustainable", R.string.axis_sustainable),
    )
}
